app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class TweetsController < ApplicationController
+#==============================tweets================================ 
+  get '/tweets' do 
+    redirect '/login' if !logged_in?
+    
+    erb :"tweets/tweets"
+  end
+#===============================new================================== 
+  get '/tweets/new' do 
+    redirect '/login' if !logged_in?
+    
+    erb :'tweets/create_tweet'
+  end
+#-------------------------------------------------------------------- 
+  post '/tweets' do 
+    redirect '/tweets/new' if empty?(params)
+    
+    Tweet.create(content: params[:content], user_id: current_user.id)
+    redirect '/tweets'
+  end
+#==============================show================================== 
+  get '/tweets/:id' do 
+    redirect '/login' if !logged_in?
+    
+    @tweet = tweet(params[:id])
+    
+    erb :'/tweets/show_tweet'
+  end
+#==============================edit================================== 
+  get '/tweets/:id/edit' do 
+    redirect '/login' if !logged_in? 
+    
+    @tweet = tweet(params[:id])
+    
+    redirect '/tweets' if @tweet.user_id != current_user.id  
```

Comment:
I don't see a lot of students checking the tweet's user id. Good job!

Reply:
:) thanks